src/core/api.proto

```diff
@@ -94,6 +94,55 @@ message InferRequestHeader
     //@@       for tensors with a non-fixed-size datatype (like STRING).
     //@@
     uint64 batch_byte_size = 3;
+
+    //@@      .. cpp:var:: message SharedMemoryElement
+    //@@
+    //@@         The Meta-data for the Input shared memory to read element from.
+    //@@
+    message SharedMemoryElement
+    {
+      //@@    .. cpp:var:: key_t shm_key
+      //@@
+      //@@     The unique key used to refer to a specific location in shared
+      //@@     memory. Optional for tensors. Required to read tensors from
+      //@@     shared memory. If provided, tensor data read from this location.
+      //@@
+      key_t shm_key = 1;
+
+      //@@    .. cpp:var:: uint64 offset
+      //@@
+      //@@     The offset for the each element.
```

Comment:
offset from the start of the shared memory region. Not for each element... this comment needs updating.